File: chrome/browser/media/router/media_source.cc
FASTVC-BENCH-CORPUS:chromium-main-v1-943b94ae-1c74-4335-94fa-ceb4d277cea8
// Copyright 2015 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "chrome/browser/media/router/media_source.h"

#include <string>

namespace media_router {

MediaSource::MediaSource(const std::string& source_id)
    : id_(source_id) {
}

MediaSource::~MediaSource() {}

std::string MediaSource::id() const {
  return id_;
}

}  // namespace media_router

File: components/media_router/browser/android/java/src/org/chromium/components/media_router/MediaSource.java
FASTVC-BENCH-CORPUS:chromium-main-v1-943b94ae-1c74-4335-94fa-ceb4d277cea8
// Copyright 2015 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package org.chromium.components.media_router;

import androidx.mediarouter.media.MediaRouteSelector;

import org.chromium.build.annotations.NullMarked;
import org.chromium.build.annotations.Nullable;

/** Abstracts parsing the Cast application id and other parameters from the source URN. */
@NullMarked
public interface MediaSource {
    /**
     * Returns a new {@link MediaRouteSelector} to use for Cast device filtering for this
     * particular media source or null if the application id is invalid.
     *
     * @return an initialized route selector or null.
     */
    public @Nullable MediaRouteSelector buildRouteSelector();

    /** @return the Cast application id corresponding to the source. */
    public String getApplicationId();

    /** @return the id identifying the media source */
    public String getSourceId();
}
